R/format_PEW_temp.R

```diff
@@ -0,0 +1,419 @@
+#'     filter(!is.na(IndvID)) %>%
+#'     #### CHECK: Remove rows without date
+#'     # filter(!is.na(Date)) %>%
+#'     #### Format and create new data columns
+#'     dplyr::select(IndvID, Species, PopID, Age, Method, BreedingSeason, Date, Sex) %>%
+#'     dplyr::rename(RingSeason = BreedingSeason) %>%
+#'     dplyr::mutate(RingAge = dplyr::case_when(Age == "0" & Method == "ChickRinging" ~ "chick",
+#'                                              Age == "1" ~ "adult",
+#'                                              Age == ">=2" ~ "adult"),
+#'                   BroodIDLaid    = NA_character_,
+#'                   BroodIDFledged = NA_character_) %>%
+#'     group_by(IndvID) %>%
+#'     dplyr::mutate(Sex_calculated = purrr::map_chr(.x = list(unique(na.omit(Sex))), .f = ~{
+#'                                                             if(length(..1) == 0){
+#'                                                               return(NA_character_)
+#'                                                             } else if(length(..1) == 1){
+#'                                                               return(..1)
+#'                                                             } else {
+#'                                                               return("C")
+#'                                                             }
+#'                                                           }),
+#'                   Sex_genetic = NA_character_) %>%
+#'     dplyr::ungroup() %>%
+#'     #### Final arrangement
+#'     dplyr::select(IndvID, Species, PopID, BroodIDLaid, BroodIDFledged,
+#'                   RingSeason, RingAge, Sex_calculated, Sex_genetic) %>%
+#'     #### Remove duplicated rows (CHECK)
+#'     dplyr::distinct()
+#'
+#'   return(Individual_data)
+#'
+#'   }
+#'
+#'
+#' #### VARIABLES
+#' # cat(Individual_data_names, sep="\n")
+#'
+#'
+#' #### --------------------------------------~
+#' #### CAPTURUE DATA                      #### in process
+#' #### --------------------------------------~
+#'
+#'
+#' # create_capture_PEW <- function(data){
+#'
+#' Capture_data <-
+#'   pew_data %>%
+#'   #### ASSUMPTION: Date is obligatory, not allowing for NAs,
+#'   # therefore we exclude observations without the Date
+#'   # (CHECK, as we may lose some information)
+#'   filter(!(is.na(Date))) %>%
+#'   #### Rename variables
+#'   dplyr::rename(CaptureDate = Date) %>%
+#'   #### Create new variables
+#'   mutate(#### ASSUMPTION: There is no captureID info, so far I use this (CHECK)
+#'          CaptureID = paste(IndvID, "CaptureNumber", sep = "_"),
+#'          #### ASSUMPTION: Some times are strange, to simplify, round to whole hour
+#'          CaptureTime = ifelse(is.na(ObservationTimeH),
+#'                               NA,
+#'                               paste(sprintf("%02d", round(ObservationTimeH, 0)),
+#'                                     "00", sep = ":")),
+#'          Sex_observed = Sex,
+#'          LocationID = NA_character_,
```

Comment:
We know the NestBox where each individual is captured, which can be used to fill in the LocationID

Reply:
Yes, I changed that.